packages/material-ui/src/Popper/Popper.test.js

```diff
@@ -123,6 +123,47 @@ describe('<Popper />', () => {
     });
   });
 
+  describe('prop: keepMounted', () => {
+    // Test case for https://github.com/mui-org/material-ui/issues/15180
+    it('should remove the transition children in the DOM when closed whilst transition status is entering', () => {
+      const children = <p>Hello World</p>;
+
+      class OpenClose extends React.Component {
+        state = {
+          open: false,
+        };
+
+        handleClick = () => {
+          this.setState({ open: true }, () => {
+            this.setState({ open: false });
```

Comment:
What is the use case for this? I would rather set it to open, assert, close it, assert. Now it's really obscured what is expected here.

Reply:
It reproduces the problem. "open, assert, close, assert" already works.

Reply:
Could we just transform it to explicitly closing it during entering with a long duration considering the callback to setState is not available anyway for function components?

Reply:
I was suggesting the usage of the `onEntered` callback in the original issue. As it turns out, we can use `onEnter`. It's the strategy the pull request uses now. I doubt the bug can be reproduced any other way than as it's done here. It's also not very important. It took a year for somebody to provide a reproduction.

Reply:
Could we test it with the repro then? Or could somebody explain what the use case is for opening a Popper and closing it in the callback?

Reply:
What do you mean by "test it with the repro"? I The use case is explained in #15180 */context*, this test copy the repro of the issue.

Reply:
Gotcha